### ISSUE TITLE
Provider produced inconsistent final plan for target groups using random string

### ISSUE THREAD

**Issue by wetjenk** [hashicorp/terraform#27898]
### Terraform Version
```
GTampi@LDNSURLAPIT65 MINGW64 /c/NEW_HOME/000_GIT_REPOS_MS/ZZZ_terraform_infra_test/plutus-infrastructure/terraform/DEV/02_loadbalancer (master)
$ terraform13.exe --version
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/aws v3.29.1
+ provider registry.terraform.io/hashicorp/random v3.1.0

Your version of Terraform is out of date! The latest version
is 0.14.7. You can update by downloading from https://www.terraform.io/downloads.html
```

### Terraform Configuration Files

```terraform
# input vars:
variable "base_health_check" {
  type = map(any)
  default = {
    port                = "traffic-port"
    protocol            = "HTTP"
    healthy_threshold   = 5
    unhealthy_threshold = 2
    interval            = 15
    timeout             = 5
    matcher             = "200"
  }
}

variable "stickiness" {
  description = "Target group cookie stickiness policy"
  type        = map(any)
  default = {
    type            = "lb_cookie"
    cookie_duration = "3600"
    enabled         = "true"
  }
}
```

```terraform
########################################################
# Target Group
########################################################

variable "name" {
  description = "Target group name"
  type        = string
}

variable "vpc_id" {
  description = "VPC ID"
  type        = string
}

variable "listener_port" {
  description = "Listener Port"
  type        = number
}

variable "stickiness" {
  description = "Target group cookie stickiness policy"
  type        = map(any)
}

variable "health_check" {
  description = "Target group health check"
  type        = map
}

variable "tags" {
  description = "Target tags"
  type        = map(any)
}

variable "protocol" {
  description = "Target group protocol. e.g. HTTP/HTTPS"
  type        = string
}

resource "random_string" "rand_alnum" {
  length      = 3
  special     = false
  upper       = true
  lower       = true
  number      = true
  min_numeric = 2
  min_upper   = 1
  keepers = {
    # Generate a new id each time we change soem vars
    name = var.name
    vpc_id = var.vpc_id
    listener_port = var.listener_port
    #stickiness = replace(replace(jsonencode(var.stickiness), "\"", ""), ":", "=") # var.stickiness
    #health_check = replace(replace(jsonencode(var.health_check), "\"", ""), ":", "=") # var.health_check
    protocol = var.protocol
    path                = lookup(var.health_check, "path", null)
    port                = lookup(var.health_check, "port", null)
    protocol            = lookup(var.health_check, "protocol", null)
    healthy_threshold   = lookup(var.health_check, "healthy_threshold", null)
    unhealthy_threshold = lookup(var.health_check, "unhealthy_threshold", null)
    interval            = lookup(var.health_check, "interval", null)
    timeout             = lookup(var.health_check, "timeout", null)
    matcher             = lookup(var.health_check, "matcher", null)
    #
    type            = lookup(var.stickiness, "type", null)
    cookie_duration = lookup(var.stickiness, "cookie_duration", null)
    enabled         = lookup(var.stickiness, "enabled", null)
  }
}

locals {
  unique_tg_name = "${var.name}-${random_string.rand_alnum.result}"
}

resource "aws_alb_target_group" "this" {
  name     = local.unique_tg_name
  vpc_id   = var.vpc_id
  port     = var.listener_port
  protocol = var.protocol

  stickiness {
    type            = lookup(var.stickiness, "type", null)
    cookie_duration = lookup(var.stickiness, "cookie_duration", null)
    enabled         = lookup(var.stickiness, "enabled", null)
  }


  health_check {
    path                = lookup(var.health_check, "path", null)
    port                = lookup(var.health_check, "port", null)
    protocol            = lookup(var.health_check, "protocol", null)
    healthy_threshold   = lookup(var.health_check, "healthy_threshold", null)
    unhealthy_threshold = lookup(var.health_check, "unhealthy_threshold", null)
    interval            = lookup(var.health_check, "interval", null)
    timeout             = lookup(var.health_check, "timeout", null)
    matcher             = lookup(var.health_check, "matcher", null)
  }

  tags = merge(var.tags, { "random_id" = local.unique_tg_name })

  lifecycle {
    create_before_destroy = true
  }
}

output "tg_arn" {
  value = aws_alb_target_group.this.arn
}

```

### Expected Behavior
Terraform should have replaced target groups with new target groups that have new names via random function

### Actual Behavior

```
module.default_targetgroup.module.loop_target_group["plutdefault"].random_string.rand_alnum: Creating...
module.listener_rules.module.loop_target_group["pricesnap"].random_string.rand_alnum: Creating...
module.listener_rules.module.loop_target_group["nasdaqproxy"].random_string.rand_alnum: Creating...
module.listener_rules.module.loop_target_group["pricesnap"].random_string.rand_alnum: Creation complete after 0s [id=6D2]
module.default_targetgroup.module.loop_target_group["plutdefault"].random_string.rand_alnum: Creation complete after 0s [id=43E]
module.listener_rules.module.loop_target_group["nasdaqproxy"].random_string.rand_alnum: Creation complete after 0s [id=S11]
module.listener_rules.module.loop_target_group["nasdaqproxy"].aws_alb_target_group.this: Creating...
module.listener_rules.module.loop_target_group["pricesnap"].aws_alb_target_group.this: Creating...
module.default_targetgroup.module.loop_target_group["plutdefault"].aws_alb_target_group.this: Creating...
module.listener_rules.module.loop_target_group["nasdaqproxy"].aws_alb_target_group.this: Creation complete after 2s [id=arn:aws:elasticloadbalancing:eu-west-1:054669460988:targetgroup/dev-plutus-nasdaqproxy-S11/e4b2b1e7672e4e80]
module.listener_rules.module.loop_target_group["pricesnap"].aws_alb_target_group.this: Creation complete after 2s [id=arn:aws:elasticloadbalancing:eu-west-1:054669460988:targetgroup/dev-plutus-pricesnap-6D2/b7a3ef2afddf57d7]
module.default_targetgroup.module.loop_target_group["plutdefault"].aws_alb_target_group.this: Creation complete after 2s [id=arn:aws:elasticloadbalancing:eu-west-1:054669460988:targetgroup/dev-plutus-plutdefault-43E/4398f8a675ebf504]

Error: Provider produced inconsistent final plan

When expanding the plan for
module.http_listener.aws_lb_listener.frontend_http_tcp to include new values
learned so far during apply, provider "registry.terraform.io/hashicorp/aws"
produced an invalid new value for .default_action[0].target_group_arn: was
cty.StringVal("arn:aws:elasticloadbalancing:eu-west-1:054669460988:targetgroup/dev-plutus-plutdefault-S89/521f8403e8d31468"),
but now
cty.StringVal("arn:aws:elasticloadbalancing:eu-west-1:054669460988:targetgroup/dev-plutus-plutdefault-43E/4398f8a675ebf504").

This is a bug in the provider, which should be reported in the provider's own
issue tracker.

Releasing state lock. This may take a few moments...

```

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`

**Comment by jbardin** [comment]
Hi @wetjenk 

Thanks for filing the issue. This appears to be a a similar situation to #25631, but I'm not immediately certain what is causing it. Can you verify that this is reproducible with a current version of Terraform?

Thanks!

**Comment by jbardin** [comment]
I'm going to close this out since we have not heard back in a while.
This is likely solved by #26470. If the error is still reproducible with the upcoming 0.15 release, please file a new issue with the information needed to reproduce it. 

Thanks!
